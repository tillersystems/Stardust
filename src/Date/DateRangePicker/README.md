# DateRangePicker

### Usage

```jsx
import DateRangePicker from 'components/Date/DateRangePicker';

// or

import { DateRangePicker } from 'components';
```

<!-- STORY -->

### Properties

- `className` - Class needed by styled component.
- `locale` - The locale to display the weekday names.
- `value` - The value of the selected date (controlled mode).
- `minDate` - The minimum selectable date.
- `maxDate` - The maximum selectable date.
- `onChange` - Handler on date changed.

| `propName`  |     propType     | defaultValue | isRequired |
| ----------- | :--------------: | :----------: | :--------: |
| `className` |     `string`     |     `''`     |     -      |
| `locale`    |     `string`     |     `en`     |     -      |
| `value`     | `luxon.Interval` |    `null`    |     -      |
| `minDate`   | `luxon.DateTime` |    `null`    |     -      |
| `maxDate`   | `luxon.DateTime` |    `null`    |     -      |
| `onChange`  |    `function`    |  `() => {}`  |     -      |

### Examples

The `DateRangePicker` can be used in two modes, in either controlled or uncontrolled mode.
In uncontrolled mode, the `DateRangePicker` manages the selected date internally (yet, the
`onChange` handler is called upon changes). In controlled mode, the selected date is based on the
given `value` prop. The `onChange` handler is called upon user changes, but the selected value won't
be affected unless the `value` prop is updated.

#### Uncontrolled example

```jsx
import { DateRangePicker } from 'components';

render() {
  return <DateRangePicker />
}
```

#### Controlled example

```jsx
import { DateRangePicker } from 'components';
import { DateTime, Interval } from 'luxon';

let value = Interval.fromDateTimes(
    DateTime.locale(),
    DateTime.locale());

handleChange = date => value = date;

render() {
  return <DateRangePicker value={value} onChange={handleChange} />
}
```
