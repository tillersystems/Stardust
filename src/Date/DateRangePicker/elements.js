import styled from 'styled-components';

export const PickerContainer = styled.div``;

export const Separator = styled.div`
  height: 100%;
  width: 0.1rem;

  background: ${({ theme: { palette } }) => palette.gray};
`;
