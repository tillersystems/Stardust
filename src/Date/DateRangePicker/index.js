import React, { PureComponent } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { DateTime, Interval } from 'luxon';

import { Button, Icon, Theme } from '../..';
import Weeks from '../Weeks';
import WeekdayNames from '../WeekdayNames';
import { Header, HeaderTitle } from '../elements';
import { PickerContainer, Separator } from './elements';

/**
 * Gets a valid current month from the given date and max date.
 *
 * @param {luxon.DateTime} date - The date.
 * @param {luxon.DateTime} maxDate - The maximum date.
 *
 * @return {luxon.DateTime}
 */
const getCurrentMonth = (date, maxDate) => {
  return date.startOf('month').minus(date.diff(maxDate.startOf('month'), 'months').months + 1);
};

/**
 * DateRangePicker
 *
 * This component is in charge of displaying a date range picker.
 *
 * @param {string} className - Styled component class name.
 * @param {string} locale - Local to use to display the weekday names.
 * @param {luxon.Interval} value - Controlled value (the selected date).
 * @param {luxon.DateTime} minDate - Minimum allowed date.
 * @param {luxon.DateTime} maxDate - Maximum allowed date.
 * @param {function} onChange - Handler of date change.
 *
 * @return {jsx}
 */
class DateRangePicker extends PureComponent {
  /** Display name. */
  static displayName = 'DateRangePicker';

  /** Prop types validation. */
  static propTypes = {
    className: PropTypes.string,
    locale: PropTypes.string,
    value: PropTypes.object,
    minDate: PropTypes.object,
    maxDate: PropTypes.object,
    onChange: PropTypes.func,
  };

  /** Default props. */
  static defaultProps = {
    className: '',
    value: null,
    minDate: null,
    maxDate: null,
    locale: 'en',
    onChange: () => {},
  };

  /** Internal state. */
  state = {
    currentMonth: DateTime.local(),
    selectedStartDate: null,
    selectedEndDate: null,
    hoveredDate: null,
  };

  /**
   * Handles mounted event of component's lifecycle.
   */
  componentDidMount() {
    const { maxDate } = this.props;

    // Sets currentMonth in componentDidMount since relies on props
    this.setState({
      ...this.state,
      currentMonth: getCurrentMonth(DateTime.local(), maxDate || DateTime.local()),
    });
  }

  /**
   * Handles click on a date.
   *
   * @param {luxon.DateTime} date - The clicked date.
   */
  handleDateClick = date => {
    const { selectedStartDate, selectedEndDate } = this.state;
    const { onChange, maxDate } = this.props;

    if (!selectedStartDate) {
      this.setState({
        ...this.state,
        currentMonth: getCurrentMonth(date, maxDate || date),
        selectedStartDate: date,
      });
    } else {
      if (!selectedEndDate) {
        const newSelectedStartDate = DateTime.min(selectedStartDate, date);
        const newSelectedEndDate = DateTime.max(selectedStartDate, date);

        this.setState(
          {
            ...this.state,
            currentMonth: getCurrentMonth(
              newSelectedStartDate,
              maxDate || newSelectedStartDate.minus({ month: 1 }),
            ),
            selectedStartDate: newSelectedStartDate,
            selectedEndDate: newSelectedEndDate,
          },
          () => {
            onChange(Interval.fromDateTimes(selectedStartDate, date));
          },
        );
      } else {
        this.setState({
          ...this.state,
          currentMonth: getCurrentMonth(date, maxDate || date),
          selectedStartDate: date,
          selectedEndDate: null,
        });
      }
    }
  };

  /**
   * Handles mouse over a date.
   *
   * @param {luxon.DateTime} date - The date.
   */
  handleDateOver = date => {
    this.setState({ ...this.state, hoveredDate: date });
  };

  /**
   * Handles click on previous month.
   */
  handlePrevMonthClick = () => {
    const { currentMonth } = this.state;
    this.setState({
      currentMonth: currentMonth.minus({ months: 1 }),
    });
  };

  /**
   * Handles click on next month.
   */
  handleNextMonthClick = () => {
    const { currentMonth } = this.state;
    this.setState({
      currentMonth: currentMonth.plus({ months: 1 }),
    });
  };

  render() {
    const { className, value, locale, minDate, maxDate } = this.props;
    const { currentMonth, selectedStartDate, selectedEndDate, hoveredDate } = this.state;

    const actualSelectedStartDate =
      selectedStartDate && !selectedEndDate
        ? selectedStartDate
        : value
        ? value.start
        : selectedStartDate;
    const actualSelectedEndDate =
      selectedStartDate && !selectedEndDate
        ? selectedStartDate
        : value
        ? value.end
        : selectedEndDate;

    const highlightedStartDate =
      selectedStartDate && selectedEndDate
        ? hoveredDate
        : hoveredDate
        ? DateTime.min(selectedStartDate || hoveredDate, selectedEndDate || hoveredDate)
        : null;
    const highlightedEndDate =
      selectedStartDate && selectedEndDate
        ? hoveredDate
        : hoveredDate
        ? DateTime.max(selectedStartDate || hoveredDate, selectedEndDate || hoveredDate)
        : null;

    const nextMonth = currentMonth.plus({ months: 1 });

    const isAfterMinDate = !minDate || currentMonth.startOf('month') >= minDate.startOf('month');
    const isBeforeMaxDate = !maxDate || nextMonth.endOf('month') <= maxDate.endOf('month');

    return (
      <div className={className} onMouseLeave={() => this.handleDateOver(null)}>
        <PickerContainer>
          <Header>
            <Button
              data-testid="pervious-month-button"
              appearance="secondary"
              size="small"
              onClick={this.handlePrevMonthClick}
              disabled={!isAfterMinDate}
            >
              <Icon color={Theme.palette.darkBlue} name="chevron-left" />
            </Button>
            <HeaderTitle>
              {currentMonth.setLocale(locale).toLocaleString({ month: 'long', year: 'numeric' })}
            </HeaderTitle>
          </Header>
          <>
            <WeekdayNames currentMonth={currentMonth} locale={locale} />
            <Weeks
              currentMonth={currentMonth}
              selectedStartDate={
                actualSelectedStartDate ? actualSelectedStartDate.startOf('day') : null
              }
              selectedEndDate={actualSelectedEndDate ? actualSelectedEndDate.endOf('day') : null}
              highlightedStartDate={
                highlightedStartDate ? highlightedStartDate.startOf('day') : null
              }
              highlightedEndDate={highlightedEndDate ? highlightedEndDate.endOf('day') : null}
              minDate={minDate}
              maxDate={maxDate}
              onDateClick={this.handleDateClick}
              onDateOver={this.handleDateOver}
            />
          </>
        </PickerContainer>

        <Separator />

        <PickerContainer>
          <Header>
            <HeaderTitle>
              {nextMonth.setLocale(locale).toLocaleString({ month: 'long', year: 'numeric' })}
            </HeaderTitle>
            <Button
              data-testid="next-month-button"
              appearance="secondary"
              size="small"
              onClick={this.handleNextMonthClick}
              disabled={!isBeforeMaxDate}
            >
              <Icon color={Theme.palette.darkBlue} name="chevron-right" />
            </Button>
          </Header>
          <>
            <WeekdayNames currentMonth={nextMonth} locale={locale} />
            <Weeks
              currentMonth={nextMonth}
              selectedStartDate={
                actualSelectedStartDate ? actualSelectedStartDate.startOf('day') : null
              }
              selectedEndDate={actualSelectedEndDate ? actualSelectedEndDate.endOf('day') : null}
              highlightedStartDate={
                highlightedStartDate ? highlightedStartDate.startOf('day') : null
              }
              highlightedEndDate={highlightedEndDate ? highlightedEndDate.endOf('day') : null}
              minDate={minDate}
              maxDate={maxDate}
              onDateClick={this.handleDateClick}
              onDateOver={this.handleDateOver}
            />
          </>
        </PickerContainer>
      </div>
    );
  }
}

export default styled(DateRangePicker)`
  display: grid;
  grid-column-gap: ${({ theme: { dimensions } }) => dimensions.medium};
  grid-template-rows: auto;
  grid-template-columns: 20rem 0.1rem 20rem;
`;
