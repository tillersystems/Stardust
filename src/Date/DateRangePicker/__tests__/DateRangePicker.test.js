import React from 'react';
import 'jest-styled-components';
import * as Luxon from 'luxon';
import { fireEvent } from 'react-testing-library';

import DateRangePicker from '..';

describe('<DatePicker />', () => {
  const mockLocalDateTime = Luxon.DateTime.fromObject({
    year: 2018,
    month: 7,
    day: 15,
    hour: 0,
    minute: 0,
    second: 0,
    zone: 'Europe/Paris',
  });

  beforeEach(() => {
    Luxon.DateTime.local = jest.fn().mockImplementation(() => mockLocalDateTime);
  });

  it('should render without a problem', () => {
    const { container } = render(<DateRangePicker minDate={mockLocalDateTime} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should select a range date', () => {
    const { getByText } = render(<DateRangePicker maxDate={mockLocalDateTime} />);

    const dayfrom = getByText('10');
    const dayTo = getByText('12');

    // Click on first range day
    fireEvent.click(dayfrom);

    // Click on last range day
    fireEvent.click(dayTo);

    expect(dayfrom).toBeInTheDocument();
    expect(dayTo).toBeInTheDocument();
  });

  it('should handle previous month', () => {
    const { getByTestId, getByText } = render(<DateRangePicker />);

    const previousMonthButton = getByTestId('previous-month-button');

    // Click on previous month button
    fireEvent.click(previousMonthButton);

    const expectedMonth = getByText(/June/);

    expect(expectedMonth).toBeInTheDocument();

    expect(render).toMatchSnapshot();
  });

  it('should handle next month', () => {
    const { getByTestId, getByText } = render(<DateRangePicker />);

    const nextMonthButton = getByTestId('next-month-button');

    // Click on previous month button
    fireEvent.click(nextMonthButton);

    const expectedMonth = getByText(/August/);

    expect(expectedMonth).toBeInTheDocument();
  });

  test('should handle mouse Leave', () => {
    const { container } = render(<DateRangePicker />);

    // Mouse Leave on DatePicker
    fireEvent.mouseLeave(container.firstChild);

    expect(container.firstChild).toBeInTheDocument();
  });
});
