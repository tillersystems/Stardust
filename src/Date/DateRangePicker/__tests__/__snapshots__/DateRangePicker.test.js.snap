// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> should handle previous month 1`] = `[Function]`;

exports[`<DatePicker /> should render without a problem 1`] = `
.c19 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  width: auto;
  padding: 0.8rem 1.6rem;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  line-height: 1.57;
  color: hsl(0,100%,100%);
  cursor: pointer;
  outline: none;
  -webkit-transition: background 200ms ease,box-shadow 200ms ease;
  transition: background 200ms ease,box-shadow 200ms ease;
  color: hsl(207,13%,45%);
  background: linear-gradient(180deg,hsl(0,100%,100%) 0%,hsl(216,45%,98%) 100%);
  border: 1px solid hsl(207,22%,90%);
  padding: 0.35rem 1.4rem;
}

.c19 svg > path {
  fill: hsl(0,100%,100%);
}

.c19 svg > path {
  fill: hsl(207,13%,45%);
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  width: auto;
  padding: 0.8rem 1.6rem;
  border-radius: 0.4rem;
  font-size: 1.4rem;
  line-height: 1.57;
  color: hsl(0,100%,100%);
  cursor: pointer;
  outline: none;
  -webkit-transition: background 200ms ease,box-shadow 200ms ease;
  transition: background 200ms ease,box-shadow 200ms ease;
  color: hsl(207,13%,45%);
  background: linear-gradient(180deg,hsl(0,100%,100%) 0%,hsl(216,45%,98%) 100%);
  border: 1px solid hsl(207,22%,90%);
  padding: 0.35rem 1.4rem;
  cursor: not-allowed;
  opacity: 0.4;
}

.c2 svg > path {
  fill: hsl(0,100%,100%);
}

.c2 svg > path {
  fill: hsl(207,13%,45%);
}

.c8 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c8 > .c9 {
  color: hsl(207,22%,90%);
}

.c8 > .c9 {
  color: hsl(207,22%,90%);
}

.c11 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
}

.c11 > .c9 {
  color: hsl(207,22%,90%);
}

.c11 > .c9 {
  color: hsl(207,22%,90%);
}

.c20 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c20 > .c9 {
  color: hsl(207,22%,90%);
}

.c13 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c13 > .c9 {
  color: hsl(207,22%,90%);
}

.c14 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
}

.c14 > .c9 {
  color: hsl(207,22%,90%);
}

.c15 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c15 > .c9 {
  color: hsl(207,22%,90%);
}

.c22 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c23 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
}

.c24 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c25 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
}

.c25 > .c9 {
  color: hsl(207,22%,90%);
}

.c26 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c26 > .c9 {
  color: hsl(207,22%,90%);
}

.c27 > .c9 {
  color: hsl(207,22%,90%);
}

.c17 {
  font-size: 1.4rem;
  line-height: 3rem;
  cursor: pointer;
  cursor: not-allowed;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c17 > .c9 {
  color: hsl(207,22%,90%);
}

.c17 > .c9 {
  color: hsl(207,22%,90%);
}

.c10 {
  width: 100%;
  height: 100%;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
}

.c12 {
  width: 100%;
  height: 100%;
}

.c21 {
  width: 100%;
  height: 100%;
  border-top-left-radius: 0.4rem;
  border-bottom-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c16 {
  width: 100%;
  height: 100%;
  border-top-right-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(7,2.8rem);
  grid-template-rows: 2.8rem;
  justify-items: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  text-align: center;
  color: hsl(206,23%,69%);
}

.c6 {
  font-size: 1.4rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(7,2.8rem);
  grid-template-rows: 2rem;
  justify-items: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  text-align: center;
  border-bottom: 1px solid hsl(206,23%,69%);
  color: hsl(206,23%,69%);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1.5rem;
  color: hsl(213,17%,20%);
  font-size: 1.4rem;
  line-height: 1.4rem;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: center;
}

.c3 {
  display: inline-block;
  border-image-source: none;
  border-image-width: 0;
  padding: 0;
  font-size: 20;
  line-height: 20;
}

.c18 {
  height: 100%;
  width: 0.1rem;
  background: hsl(206,23%,69%);
}

.c0 {
  display: grid;
  grid-column-gap: 1.5rem;
  grid-template-rows: auto;
  grid-template-columns: 20rem 0.1rem 20rem;
}

<div
  class="c0"
>
  <div
    class=""
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="previous-month-button"
        disabled=""
        type="button"
      >
        <div
          class="c3"
          data-testid="iconContainer"
          height="20"
          width="20"
        >
          <svg
            data-testid="iconSvg"
            height="20"
            style="display: block;"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M335.085 368.915c8.331 8.331 8.331 21.839 0 30.17-8.331 8.331-21.839 8.331-30.17 0l-128-128c-8.331-8.331-8.331-21.839 0-30.17l128-128c8.331-8.331 21.839-8.331 30.17 0 8.331 8.331 8.331 21.839 0 30.17L222.17 256l112.915 112.915z"
              data-testid="iconSvgPath"
              fill="hsl(213,17%,20%)"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
      <div
        class="c4"
      >
        June 2018
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        Mon
      </div>
      <div
        class="c6"
      >
        Tue
      </div>
      <div
        class="c6"
      >
        Wed
      </div>
      <div
        class="c6"
      >
        Thu
      </div>
      <div
        class="c6"
      >
        Fri
      </div>
      <div
        class="c6"
      >
        Sat
      </div>
      <div
        class="c6"
      >
        Sun
      </div>
    </div>
    <div>
      <div
        class="c7"
      >
        <div
          class="c8"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            28
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            29
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            30
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            31
          </div>
        </div>
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            01
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            02
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            03
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            04
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            05
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            06
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            07
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            08
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            09
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            11
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            12
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            13
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            14
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            15
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            16
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            17
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            18
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            19
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            20
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            21
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            22
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            23
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            24
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            25
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            26
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            27
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            28
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            29
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            30
          </div>
        </div>
        <div
          class="c17"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            01
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c18"
  />
  <div
    class=""
  >
    <div
      class="c1"
    >
      <div
        class="c4"
      >
        July 2018
      </div>
      <button
        class="c19"
        data-testid="next-month-button"
        type="button"
      >
        <div
          class="c3"
          data-testid="iconContainer"
          height="20"
          width="20"
        >
          <svg
            data-testid="iconSvg"
            height="20"
            style="display: block;"
            viewBox="0 0 512 512"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M176.915 143.085c-8.331-8.331-8.331-21.839 0-30.17 8.331-8.331 21.839-8.331 30.17 0l128 128c8.331 8.331 8.331 21.839 0 30.17l-128 128c-8.331 8.331-21.839 8.331-30.17 0-8.331-8.331-8.331-21.839 0-30.17L289.83 256 176.915 143.085z"
              data-testid="iconSvgPath"
              fill="hsl(213,17%,20%)"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        Mon
      </div>
      <div
        class="c6"
      >
        Tue
      </div>
      <div
        class="c6"
      >
        Wed
      </div>
      <div
        class="c6"
      >
        Thu
      </div>
      <div
        class="c6"
      >
        Fri
      </div>
      <div
        class="c6"
      >
        Sat
      </div>
      <div
        class="c6"
      >
        Sun
      </div>
    </div>
    <div>
      <div
        class="c7"
      >
        <div
          class="c8"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            25
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            26
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            27
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            28
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            29
          </div>
        </div>
        <div
          class="c11"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            30
          </div>
        </div>
        <div
          class="c20"
          disabled=""
        >
          <div
            class="c9 c21"
          >
            01
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            02
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            03
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            04
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            05
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            06
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            07
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            08
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c13"
          disabled=""
        >
          <div
            class="c9 c10"
          >
            09
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            10
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            11
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            12
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            13
          </div>
        </div>
        <div
          class="c14"
          disabled=""
        >
          <div
            class="c9 c12"
          >
            14
          </div>
        </div>
        <div
          class="c15"
          disabled=""
        >
          <div
            class="c9 c16"
          >
            15
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c22"
        >
          <div
            class="c9 c10"
          >
            16
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            17
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            18
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            19
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            20
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            21
          </div>
        </div>
        <div
          class="c24"
        >
          <div
            class="c9 c16"
          >
            22
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c22"
        >
          <div
            class="c9 c10"
          >
            23
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            24
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            25
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            26
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            27
          </div>
        </div>
        <div
          class="c23"
        >
          <div
            class="c9 c12"
          >
            28
          </div>
        </div>
        <div
          class="c24"
        >
          <div
            class="c9 c16"
          >
            29
          </div>
        </div>
      </div>
      <div
        class="c7"
      >
        <div
          class="c22"
        >
          <div
            class="c9 c10"
          >
            30
          </div>
        </div>
        <div
          class="c24"
        >
          <div
            class="c9 c16"
          >
            31
          </div>
        </div>
        <div
          class="c25"
        >
          <div
            class="c9 c12"
          >
            01
          </div>
        </div>
        <div
          class="c25"
        >
          <div
            class="c9 c12"
          >
            02
          </div>
        </div>
        <div
          class="c25"
        >
          <div
            class="c9 c12"
          >
            03
          </div>
        </div>
        <div
          class="c25"
        >
          <div
            class="c9 c12"
          >
            04
          </div>
        </div>
        <div
          class="c26"
        >
          <div
            class="c9 c16"
          >
            05
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
